sql/core/src/test/scala/org/apache/spark/sql/DatasetPrimitiveSuite.scala

```diff
@@ -28,6 +28,8 @@ case class SeqClass(s: Seq[Int])
 
 case class ListClass(l: List[Int])
 
+case class JListClass(l: java.util.List[Int])
```

Comment:
I did not notice that there is separate inference code for Java classes. It would certainly be nice to unify the code for Java and Scala classes. Is this already being worked on/planned?

I moved the `List` support to `JavaTypeInference` and rewrote tests accordingly.